p2p/libp2p/networksharding/listsSharder.go

```diff
@@ -252,6 +254,11 @@ func (ls *listsSharder) splitPeerIds(peers []peer.ID) map[int]sorting.PeerDistan
 		peerInfo := ls.peerShardResolver.GetPeerInfo(pid)
 		ls.mutResolver.RUnlock()
 
+		ls.preferredPeersHolder.Put(peerInfo.PkBytes, pid, peerInfo.ShardID)
```

Comment:
please add a TODO here to think if we need to move this elsewhere

Reply:
done